sdk/tables/azure-data-tables/src/table_clients.cpp

```diff
@@ -541,7 +541,7 @@ Azure::Response<Models::DeleteEntityResult> TableClient::DeleteEntity(
 {
   auto url = m_url;
   url.AppendPath(
-      m_tableName + PartitionKeyFragment
+      Azure::Core::Url::Encode(m_tableName) + PartitionKeyFragment
```

Comment:
Same comment as before about Encode(a)+Encode(b)+Encode(c).

Note: I'm assuming that `Encode(PartitionKeyFragment)==PartitionKeyFragment` - in other words that PartitionKeyFragment (and RowKeyFragment) won't be changed by URL encoding (and thus that they wouldn't be affected by URL encoding).

Reply:
done

Reply:
> Note: I'm assuming that `Encode(PartitionKeyFragment)==PartitionKeyFragment` - in other words that PartitionKeyFragment (and RowKeyFragment) won't be changed by URL encoding (and thus that they wouldn't be affected by URL encoding).

That assumption doesn't hold. Both the partition and row key fragments have `=` which does get encoded by default.
```C++
std::string PartitionKeyFragment = "(PartitionKey='";

 // Assertion fails, encoded string is "%28PartitionKey%3D%27"
EXPECT_EQ(Azure::Core::Url::Encode(PartitionKeyFragment), PartitionKeyFragment);
```